### PR TITLE
Add __repr__ method to InvalidRow for improved debugging

### PR DESCRIPTION
**Pull request by GreatBahram**
**Problem**

What problem have you solved?
This commit introduces a `__repr__` method for InvalidRow, allowing exceptions to be more easily inspected during debugging.

**Solution**
Adding `__repr__` method :shrug: 

How did you solve the problem?

**Acceptance Criteria**

Have you written tests? Have you included screenshots of your changes if applicable?
Did you document your changes?

### CODE REVIEW

File: import_export/results.py [hunk 1]
```diff
@@ -176,6 +176,13 @@ def error_count(self):
             count += len(error_list)
         return count
 
+    def __repr__(self):
+        return (
+            f"<{type(self).__name__}(row={self.number}, "
```
Comment (GreatBahram) [inline import_export/results.py:181]: > [!NOTE]
> Here I assumed the number is representing the row, right?

Comment (matthewhegarty) [reply import_export/results.py:181]: yes, it is the row number.

File: tests/core/tests/test_results.py [hunk 2]
```diff
@@ -43,6 +43,18 @@ def test_traceback(self):
...
 
+class InvalidRowTest(SimpleTestCase):
+    def test_repr(self):
+        try:
+            raise ValidationError(message="invalid row")
+        except ValidationError as exc:
+            error = InvalidRow(validation_error=exc, number=1, values={})
+
+        self.assertEqual(
+            repr(error), "<InvalidRow(row=1, error='invalid row', error_count=1)>"
```
Comment (matthewhegarty) [inline tests/core/tests/test_results.py:54]: You need to change this line to the following for the test to pass:

```python
        self.assertEqual(
            repr(error),
            (
                "<InvalidRow(row=1, error=ValidationError(['invalid row']), "
                "error_count=1)>"
            ),
        )
```

**Review by matthewhegarty — CHANGES_REQUESTED** [review]
Hi - if you can fix the test as suggested I will merge.
thanks

File: import_export/results.py (hunk 1 above)
Comment (GreatBahram) [reply import_export/results.py:181]: Thanks :sunflower:

**Review by matthewhegarty — APPROVED** [review]

**Comment by matthewhegarty** [comment]
Thanks for submitting

**Pull request merged**